### PR TITLE
184781969 sort versions by stake

### PR DESCRIPTION
**Pull request by maciejkocylapc**
#### What's this PR do?
- sort software versions by stake

#### How should this be manually tested?
- run tests
- check if software versions on home page are sorted by stake percent correctly

#### What are the relevant tickets?
- [URL to the PivotalTracker ticket](https://www.pivotaltracker.com/story/show/184781969)

#### Task completed checklist:
- [] Is there appropriate test coverage?
- [] Did I take into consideration possible security vulnerabilities?
- [] Are the controllers secure?
- [] Was every important matter documented?

#### Things to watch out for
- [N] Does this PR have any migrations?
- [N] Does this PR add new dependencies?

### CODE REVIEW

File: app/helpers/validators_helper.rb [hunk 1]
```diff
@@ -75,7 +75,7 @@ def skipped_vote_percent(validator, batch)
   end
 
   def sort_software_versions(versions)
-    versions&.sort_by { |ver| Gem::Version.new(ver.keys.first)}&.reverse
+    versions&.sort_by { |ver| ver.values[0]["stake_percent"] }&.reverse
```
Comment (kamtarnowski) [inline app/helpers/validators_helper.rb:78]: There are two negative scenarios: 
1) versions are invalid and AppSignal will send an alert (we would notice that faster and think about some fix)
In this case I would suggest to make a change 
```suggestion
    versions.sort_by { |ver| ver.values[0]["stake_percent"] }.reverse
```

2) versions are invalid (e.g. nil) and we do not Rails to raise any problem (software versions list will be blank on homepage)
```suggestion
    Array(versions).sort_by { |ver| ver.values[0]["stake_percent"] }&.reverse
```

Comment (kamtarnowski) [reply app/helpers/validators_helper.rb:78]: IMO we should stick with the 1st scenario.

Comment (maciejkocylapc) [reply app/helpers/validators_helper.rb:78]: I do not think this is a good idea. Raising an error will crash the front stats daemon and we don't want that. If something goes wrong we will still be able to see it via blank software versions component

Comment (kamtarnowski) [reply app/helpers/validators_helper.rb:78]: Yeah, I haven't notice that this is running in front stats daemon.

**Review by kamtarnowski — APPROVED** [review]

**Review by kbiala — APPROVED** [review]

**Pull request merged**